src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1180,17 +1180,17 @@ MethodTableBuilder::bmtInterfaceEntry::CreateSlotTable(
 
     CONSISTENCY_CHECK(m_pImplTable == NULL);
 
-    SLOT_INDEX cSlots = (SLOT_INDEX)GetInterfaceType()->GetMethodTable()->GetNumVirtuals();
+    MethodTable *interfaceMT = GetInterfaceType()->GetMethodTable();
+    SLOT_INDEX cSlots = interfaceMT->GetNumVirtuals();
+    if (interfaceMT->HasVirtualStaticMethods())
+    {
+        cSlots += interfaceMT->GetNumNonVirtualSlots();
+    }
     bmtInterfaceSlotImpl * pST = new (pStackingAllocator) bmtInterfaceSlotImpl[cSlots];
 
     MethodTable::MethodIterator it(GetInterfaceType()->GetMethodTable());
-    for (; it.IsValid(); it.Next())
```

Comment:
I'd prefer to see the iterator as it is, and just check it.IsStatic and break for that case as well as the !Virtual case.

Reply:
Well, the problem is that the IsVirtual check is actually incorrect in this case - it just checks that the slot index is less then <code>GetNumVirtuals</code>.

Reply:
One other thing that seems somewhat concerning to me is that, when SVMs are present, we create the slot table for all non-virtual methods. We could easily place SVMs first followed by the other non-virtual methods and have another counter (<code>GetNumVirtualInstanceAndStaticMethods</code> or something) to account for the two subsets we need to generate the slot table for (it is used in <code>FindDeclMethodOnInterfaceEntry</code>).

Reply:
Fixed per our offline chat by reverting the changes to CreateSlotTable and short-circuiting ProcessMethodImpls in the presence of SVMs.